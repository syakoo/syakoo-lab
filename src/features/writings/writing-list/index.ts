export { WritingList } from "./writing-list";
