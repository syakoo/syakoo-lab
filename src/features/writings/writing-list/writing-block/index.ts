export { WritingBlock } from "./writing-block";
