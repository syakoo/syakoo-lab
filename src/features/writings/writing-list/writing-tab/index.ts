export { WritingTab } from "./writing-tab";
