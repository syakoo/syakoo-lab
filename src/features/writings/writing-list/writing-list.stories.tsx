import type { Meta, StoryObj } from "@storybook/nextjs";
import { clearAllMocks, mocked } from "storybook/test";

import { generateDummyWritingHead } from "@/entities/writing/models/writing.mocks";
import { readWritingHeads } from "@/features/writings/writing-reader";
import { random } from "@/shared/test-utils/random/random";
import { range } from "@/shared/utils/array/range";

import { WritingList } from ".";

const meta = {
  component: WritingList,
  parameters: {
    layout: "fullscreen",
    testLevel: "snapshot",
  },
  afterEach: () => {
    clearAllMocks();
  },
} satisfies Meta<typeof WritingList>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Sample: Story = {
  beforeEach: () => {
    const mockHeads = range(0, random.integer(5, 10)).map(() =>
      generateDummyWritingHead(),
    );
    mocked(readWritingHeads).mockResolvedValue(mockHeads);
  },
};
