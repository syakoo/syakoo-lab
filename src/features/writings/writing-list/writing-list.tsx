import { compareDesc } from "date-fns";

import { readWritingContents } from "@/contents/writings/reader";
import { resolveWritingHead } from "@/features/writings/_models/head-resolver";

import { WritingListView } from "./writing-list.view";

export const WritingList = async () => {
  const metas = (await readWritingContents())
    .map(({ frontMatter }) => resolveWritingHead(frontMatter))
    .sort((left, right) =>
      compareDesc(new Date(left.published), new Date(right.published)),
    );

  return <WritingListView heads={metas} />;
};
