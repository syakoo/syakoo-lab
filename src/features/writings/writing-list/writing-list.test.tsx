import { testStories } from "@/test/test-stories";

import * as stories from "./writing-list.stories";

describe("WritingList", () => {
  testStories(stories);
});
