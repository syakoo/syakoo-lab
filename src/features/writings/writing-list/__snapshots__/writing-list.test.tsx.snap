// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WritingList > スナップショット Sample 1`] = `
<body>
  <div>
    <section>
      <div
        class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_column__zqca9tk flex_flexStyle_gap_300__zqca9t10 flex_flexStyle_variants_gap_300__zqca9tf flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
      >
        <div
          class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_column__zqca9tk flex_flexStyle_gap_100__zqca9ty flex_flexStyle_variants_gap_100__zqca9td flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
        >
          <h2
            class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_500__1djq151p text_textStyle_variants_size_500__1djq151a text_textStyle_weight_bold__1djq151t"
          >
            Writings
          </h2>
          <div
            class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_100__zqca9ty flex_flexStyle_variants_gap_100__zqca9td flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
          >
            <a
              class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3"
              href="/writings"
            >
              <div
                aria-current="true"
                class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_true__1g56vq61"
              >
                All
              </div>
            </a>
            <a
              class="link_linkStyle__1vz8ftz0"
              href="/writings?type=article"
            >
              <div
                aria-current="false"
                class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_false__1g56vq62"
              >
                Article
              </div>
            </a>
            <a
              class="link_linkStyle__1vz8ftz0"
              href="/writings?type=note"
            >
              <div
                aria-current="false"
                class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_false__1g56vq62"
              >
                Note
              </div>
            </a>
            <a
              class="link_linkStyle__1vz8ftz0"
              href="/writings?type=diary"
            >
              <div
                aria-current="false"
                class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_false__1g56vq62"
              >
                Diary
              </div>
            </a>
          </div>
        </div>
        <ul
          class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_column__zqca9tk flex_flexStyle_gap_200__zqca9tz flex_flexStyle_variants_gap_200__zqca9te flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
        >
          <li
            class="fade-in_fadeInStyle__lraw9s1"
            style="animation-duration: 0.5s; animation-delay: 0s;"
          >
            <article
              class="flex_flexStyle__zqca9ti flex_flexStyle_align_center__zqca9ts flex_flexStyle_variants_align_center__zqca9t7 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_200__zqca9tz flex_flexStyle_variants_gap_200__zqca9te flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
            >
              <div
                class="writing-block_writingBlockStyles_iconWrapper__1nw36u70"
              >
                <svg
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.5 14.25v-2.625a3.375 3.375 0 00-3.375-3.375h-1.5A1.125 1.125 0 0113.5 7.125v-1.5a3.375 3.375 0 00-3.375-3.375H8.25m0 12.75h7.5m-7.5 3H12M10.5 2.25H5.625c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125V11.25a9 9 0 00-9-9z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
              <div>
                <a
                  class="link_linkStyle__1vz8ftz0"
                  href="/writings/sample-writing-meta-id1"
                >
                  <h3
                    class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_200__1djq151m text_textStyle_variants_size_200__1djq1517 text_textStyle_weight_bold__1djq151t"
                  >
                    Sample Article
                  </h3>
                </a>
                <p
                  class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                >
                  2023-10-09
                </p>
                <div
                  class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
                >
                  <p
                    class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                  >
                    #
                    雑記
                  </p>
                  <p
                    class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                  >
                    #
                    Sample
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="fade-in_fadeInStyle__lraw9s1"
            style="animation-duration: 0.5s; animation-delay: 0.05s;"
          >
            <article
              class="flex_flexStyle__zqca9ti flex_flexStyle_align_center__zqca9ts flex_flexStyle_variants_align_center__zqca9t7 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_200__zqca9tz flex_flexStyle_variants_gap_200__zqca9te flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
            >
              <div
                class="writing-block_writingBlockStyles_iconWrapper__1nw36u70"
              >
                <svg
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L10.582 16.07a4.5 4.5 0 01-1.897 1.13L6 18l.8-2.685a4.5 4.5 0 011.13-1.897l8.932-8.931zm0 0L19.5 7.125M18 14v4.75A2.25 2.25 0 0115.75 21H5.25A2.25 2.25 0 013 18.75V8.25A2.25 2.25 0 015.25 6H10"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
              <div>
                <a
                  class="link_linkStyle__1vz8ftz0"
                  href="/writings/sample-writing-meta-id2"
                >
                  <h3
                    class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_200__1djq151m text_textStyle_variants_size_200__1djq1517 text_textStyle_weight_bold__1djq151t"
                  >
                    Sample Note
                  </h3>
                </a>
                <p
                  class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                >
                  2023-10-10
                </p>
                <div
                  class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
                >
                  <p
                    class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                  >
                    #
                    雑記
                  </p>
                  <p
                    class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                  >
                    #
                    Sample
                  </p>
                </div>
              </div>
            </article>
          </li>
          <li
            class="fade-in_fadeInStyle__lraw9s1"
            style="animation-duration: 0.5s; animation-delay: 0.1s;"
          >
            <article
              class="flex_flexStyle__zqca9ti flex_flexStyle_align_center__zqca9ts flex_flexStyle_variants_align_center__zqca9t7 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_200__zqca9tz flex_flexStyle_variants_gap_200__zqca9te flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
            >
              <div
                class="writing-block_writingBlockStyles_iconWrapper__1nw36u70"
              >
                <svg
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 6.042A8.967 8.967 0 006 3.75c-1.052 0-2.062.18-3 .512v14.25A8.987 8.987 0 016 18c2.305 0 4.408.867 6 2.292m0-14.25a8.966 8.966 0 016-2.292c1.052 0 2.062.18 3 .512v14.25A8.987 8.987 0 0018 18a8.967 8.967 0 00-6 2.292m0-14.25v14.25"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
              <div>
                <a
                  class="link_linkStyle__1vz8ftz0"
                  href="/writings/sample-writing-meta-id3"
                >
                  <h3
                    class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_200__1djq151m text_textStyle_variants_size_200__1djq1517 text_textStyle_weight_bold__1djq151t"
                  >
                    Sample Diary
                  </h3>
                </a>
                <p
                  class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                >
                  2023-10-11
                </p>
                <div
                  class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
                >
                  <p
                    class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
                  >
                    #
                    雑記
                  </p>
                </div>
              </div>
            </article>
          </li>
        </ul>
      </div>
    </section>
  </div>
</body>
`;
